### PR TITLE
feat(tiering): simple offload loop

### PR DESCRIPTION
**Pull request by dranikpg**
Simplest offloading loop, once memory budget was exceed it kicks in

### CODE REVIEW

File: src/server/tiered_storage.cc [hunk 1]
```diff
@@ -22,10 +22,14 @@
 #include "server/tiering/common.h"
 #include "server/tiering/op_manager.h"
 #include "server/tiering/small_bins.h"
+#include "server/tx_base.h"
 
 ABSL_FLAG(bool, tiered_storage_cache_fetched, true,
           "WIP: Load results of offloaded reads to memory");
 
+ABSL_FLAG(size_t, tiered_storage_max_stashes, 50,
```
Comment (romange) [inline src/server/tiered_storage.cc:30]: `tiered_storage_max_stashes` -> ` tiered_storage_write_depth`,

the term used is queue depth, so `tiered_storage_write_depth` is more clear

File: src/server/server_family.cc [hunk 2]
```diff
@@ -2083,9 +2083,13 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
   }
 
   if (should_enter("TIERED", true)) {
+    append("tiered_entries_cnt", total.tiered_entries);
+    append("tiered_entries_bytes", total.tiered_size);
```
Comment (romange) [inline src/server/server_family.cc:2087]: `tiered_size` -> `tiered_used_bytes`

File: src/server/tiered_storage.cc [hunk 3]
```diff
@@ -264,4 +293,33 @@ TieredStats TieredStorage::GetStats() const {
...
+      absl::GetFlag(FLAGS_tiered_storage_max_stashes) - op_manager_->GetStats().pending_stash_cnt;
+  if (stash_limit <= 0)
+    return;
+
+  auto cb = [this, dbid, &stash_limit](PrimeIterator it) {
+    if (it->second.HasIoPending() || it->second.IsExternal())
+      return;
+
+    if (ShouldStash(it->second)) {
+      std::string tmp;
```
Comment (romange) [inline src/server/tiered_storage.cc:308]: consider pulling tmp out of cb, to reuse it for several invocations

File: src/server/tiered_storage.cc [hunk 4]
```diff
@@ -264,4 +293,33 @@ TieredStats TieredStorage::GetStats() const {
   return stats;
 }
 
+void TieredStorage::RunOffloading(DbIndex dbid) {
```
Comment (romange) [inline src/server/tiered_storage.cc:296]: does anyone call it?

**Review by romange — COMMENTED** [review]
Nice. I do not remember what was decided about the offloading policy. @adiholden 
I remember discussing it, do we have it in the design document?

File: src/server/server_family.cc [hunk 5]
```diff
@@ -2083,9 +2083,13 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
   }
 
   if (should_enter("TIERED", true)) {
+    append("tiered_entries_cnt", total.tiered_entries);
```
[the remark this answers is not in the document]
Comment (dranikpg) [reply src/server/server_family.cc:2086]: We should have some guidelines for naming, I use _cnt below as well

**Comment by dranikpg** [comment]
> I do not remember what was decided about the offloading policy

Currently I just copied what the old tiering code was doing more or less

**Review by romange — APPROVED** [review]

**Pull request merged**